flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/ArrayElementOutputTypeStrategy.java

```diff
@@ -1,47 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.flink.table.types.inference.strategies;
-
-import org.apache.flink.annotation.Internal;
-import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
-import org.apache.flink.table.types.CollectionDataType;
-import org.apache.flink.table.types.DataType;
-import org.apache.flink.table.types.inference.CallContext;
-import org.apache.flink.table.types.inference.TypeStrategy;
-import org.apache.flink.table.types.logical.LogicalTypeRoot;
-
-import java.util.Optional;
-
-/** Specific {@link TypeStrategy} for {@link BuiltInFunctionDefinitions#ARRAY_MAX}. */
-@Internal
-public class ArrayElementOutputTypeStrategy implements TypeStrategy {
-    @Override
-    public Optional<DataType> inferType(CallContext callContext) {
-        DataType inputDataType = callContext.getArgumentDataTypes().get(0);
-        if (inputDataType.getLogicalType().getTypeRoot() != LogicalTypeRoot.ARRAY) {
-            return Optional.empty();
-        }
-        final DataType elementDataType = ((CollectionDataType) inputDataType).getElementDataType();
-        if (inputDataType.getLogicalType().isNullable()) {
```

Comment:
comment:
we don't need it.

Reply:
why don't change the code base on this class, and you can also you my test class to test your TypeStrategy in [SpecificTypeStrategies.java](https://github.com/apache/flink/pull/22917/files/20f5fe4261cefe2e3e52405489833aafb41c2252#diff-8e19791dcf467c049d2e451a8348222ab206dcf128105edc2f55adc926d120d6)

Reply:
I think you cannot remove this class, because this class name is ArrayElementOutputTypeStrategy, you must to check whether the input is a Array. if other one use this strategy but their input strategy is not an array, this class can throw exception.